### PR TITLE
Initial pass on scaffolding docs

### PR DESCRIPTION
**Pull request by jameswburke**
## Issue(s): Relates to or closes...
IRV-515

## Summary: This pull request will...
Scaffold the initial structure of Storybook documentation.

## Tests: I know this code works because...
Already released to http://storybook.irvingjs.com/

### CODE REVIEW

File: stories/05-architecture/07-Component-Mapping.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+A React component can map to more than one API component. Each API component can map to only one React component.
+
+# Component Mapping
+
+Irving maps components in the API to React components on the frontend via a component map (`componentMap.js`). Irving core includes a prebuilt component map to get you started. Components will not render unless they appear in the component map.
+
+To map an API component to a React component, identify the name of each component and add them as a key-value pair to the component map.
+
+## Identifying the name of an API component
```
Comment (ostowe) [inline stories/05-architecture/07-Component-Mapping.stories.mdx:73]: Looks like you've got the content of this section duplicated.

File: stories/05-architecture/06-Component-Endpoint.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+{
+  "defaults": [],
+  "page": [],
+  "providers": [],
+  "redirectTo": "",
+  "redirectStatus": 0
+}
+```
+
+* `defaults`: An array of components that informs Irving what components are "global" and should persist across all requests. This is good for setting up headers, footers, scripts, and other components that should exist on every page. Defaults can be passed at any point, but Irving will automatically add `context=site` to the initial request, which can be used as a flag to return the default components.
```
Comment (ostowe) [inline stories/05-architecture/06-Component-Endpoint.stories.mdx:22]: Could probably add a bit more flavor here about what this means exactly. Maybe something like "This means components appearing in the `defaults` array will only render once on the initial, server-side load of your Irving site"

File: stories/05-architecture/06-Component-Endpoint.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+* `page`: An array of components for a specific route. Irving looks for top-level conflicts between the `defaults` and `page` arrays and overrides the defaults with the page component.
+* `providers`: An array of components that should be used as providers. See the [RootProvider](https://github.com/alleyinteractive/irving/tree/production/components/rootProviders) component for full documentation.
+* `redirectTo`: The url you want to redirect to.
+* `redirectStatus`: HTTP status code for your redirect.
+
+[Keep Reading - Creating and understanding your first component](Creating-and-understanding-your-first-component)
```
Comment (ostowe) [inline stories/05-architecture/06-Component-Endpoint.stories.mdx:28]: Does this still lead somewhere?

Comment (jameswburke) [reply stories/05-architecture/06-Component-Endpoint.stories.mdx:28]: No idea, it'll need to be edited later

File: stories/05-architecture/06-Component-Endpoint.stories.mdx (hunk 2 above)
Comment (jameswburke) [reply stories/05-architecture/06-Component-Endpoint.stories.mdx:22]: Copy paste job, we'll edit later.

**Pull request merged**